src/oom/oomd-util.c

```
@@ -232,11 +232,20 @@ int oomd_sort_cgroup_contexts(Hashmap *h, oomd_compare_t compare_func, const cha
 }
 
 int oomd_cgroup_kill(const char *path, bool recurse, bool dry_run) {
-        _cleanup_set_free_ Set *pids_killed = NULL;
         int r;
 
         assert(path);
 
+        /* First try to send SIG0 recursively to ensure all child cgroups can be killed. */
```

What happens in the unlikely case that there is an unkillable process directly in `path`?
Isn't better to keep the check in the for loop, iterating on children only?

---

Hmm? I think this commit mostly does not change the behavior. Only behavior change is oomd_kill_by_pgscan_rate() and friends now may propagate error in sending SIG0, but the caller of them anyway ignore the errors except for -ENOMEM.

So, if there exist some processes that cannot be killed, then this skip killing processes in the cgroup, and try to next one. That should be the same as what you made.

Or, do I misunderstand what you mean?

---

I see now, you moved the check down, in `oomd_cgroup_kill()`